Userland/Libraries/LibJS/Lexer.cpp

```diff
@@ -310,21 +333,55 @@ bool Lexer::is_line_terminator() const
 {
     if (m_current_char == '\n' || m_current_char == '\r')
         return true;
-    if (m_position > 0 && m_position + 1 < m_source.length()) {
-        auto three_chars_view = m_source.substring_view(m_position - 1, 3);
-        return (three_chars_view == LINE_SEPARATOR) || (three_chars_view == PARAGRAPH_SEPARATOR);
-    }
-    return false;
+    if (!is_unicode_character())
+        return false;
+
+    auto codepoint = current_codepoint();
+    return codepoint == LINE_SEPARATOR || codepoint == PARAGRAPH_SEPARATOR;
+}
+
+bool Lexer::is_unicode_character() const
+{
+    return (m_current_char & 128) != 0;
+}
+
+u32 Lexer::current_codepoint() const
+{
+    static constexpr const u32 REPLACEMENT_CHARACTER = 0xFFFD;
+    if (m_position == 0)
+        return REPLACEMENT_CHARACTER;
+    Utf8View utf_8_view {m_source.substring_view(m_position - 1)};
+    return *utf_8_view.begin();
```

Comment:
Using `Utf8View` here since it can decode but I don't know if this is the best way to do this